test/e2e/next-font/with-proxy.test.ts

```diff
@@ -0,0 +1,45 @@
+import { createNextDescribe, FileRef } from 'e2e-utils'
+import { check } from 'next-test-utils'
+import { join } from 'path'
+
+createNextDescribe(
+  'next/font/google with proxy',
+  {
+    skipStart: true,
+    files: new FileRef(join(__dirname, 'with-proxy')),
+    dependencies: {
+      react: 'latest',
+      'react-dom': 'latest',
+      'https-proxy-agent': '5.0.1',
+    },
+    skipDeployment: true,
+    env: {
+      https_proxy: 'https://localhost:999999',
```

Comment:
Would be nice to spin up a local `http.Server` on a random port that just proxies the request with `http-proxy` to verify the agent is fully used correctly

Reply:
Changed the test to spin up a proxy server and a mock server. Then it asserts that the requests has reached the target server for both CSS fetches and font file fetches.